libs/libc/netdb/lib_dnsquery.c

```diff
@@ -709,6 +709,14 @@ int dns_query(FAR const char *hostname, FAR union dns_addr_u *addr,
   FAR struct dns_query_s query;
   int ret;
 
+  /* Has the DNS client been properly initialized? */
+
+  if (!dns_initialize())
```

Comment:
we have a better patch to fix this problem. @zhhyu7 please provide the patch.

Reply:
@curuvar please review the patch here: https://github.com/apache/incubator-nuttx/pull/6841